### PR TITLE
Proxy Policies

### PR DESCRIPTION
**Pull request by refs**
# What?
Route the user to X or Y backend depending on the accounts service.

## How?

`ocis-proxy` config allows to specify a policy for a given user. By fetching the user info from the accounts service and deciding where to route the user to.

```json
{
    "policies": [
        {
            "name": "reva",
            "routes": [
                {
                    "endpoint": "/",
                    "backend": "http://localhost:9100"
                },
                {
                    "endpoint": "/.well-known/",
                    "backend": "http://localhost:9130"
                }
            ]
        },
        {
            "name": "oc10",
            "routes": [
                {
                    "endpoint": "/",
                    "backend": "http://localhost:9100"
                },
                {
                    "endpoint": "/.well-known/",
                    "backend": "http://localhost:9131"
                }
            ]
        }
    ]
}
```

so regarding the user policy we use a redirect set or another.

# Note

This depends on an update of `ocis-pkg` that adds the handler option to the service, at the moment I override the import, but this PR MUST be merged first for the proxy to work.

### ideas from

http://blog.utahcon.com/posts/reverse-proxy/

### CODE REVIEW

File: pkg/command/server.go [hunk 1]
```diff
@@ -233,3 +232,72 @@ func Server(cfg *config.Config) *cli.Command {
...
 	}
 }
+
+// NewMultiHostReverseProxy undocummented
+func NewMultiHostReverseProxy(conf *config.Config) *ReverseProxy {
+	reverseProxy := &ReverseProxy{Directors: make(map[string]map[string]func(req *gohttp.Request))}
+
+	for _, target := range conf.Routes {
+		uri, err := url.Parse(target.Backend)
+		if err != nil { /* do something with err */
```
Comment (IljaN) [inline pkg/command/server.go]: log.Fatalf here to abort initialization and inform about parsing error.

**Comment by refs** [comment]
depends on https://github.com/owncloud/ocis-pkg/pull/33

**Review by butonic — APPROVED** [review]

**Comment by IljaN** [comment]
@refs Please squash or interactive rebase next time to keep a clean commit history. :)

**Comment by refs** [comment]
phone merge :D

**Pull request merged**